core/invoker/src/main/scala/whisk/core/container/ContainerUtils.scala

```diff
@@ -164,14 +164,19 @@ trait ContainerUtils {
             val buffer = java.nio.ByteBuffer.allocate(remain)
             while (remain > 0) {
                 val read = channel.read(buffer)
-                if (read > 0)
-                    remain = read - read.toInt
+                if (read >= 0) {
+                    remain -= read
+                } else {
+                    // read < 0 means premature end of file => terminate loop
+                    remain = 0
+                }
```

Comment:
Hmm. I'm not seeing my comments here. I had asked for the same. I also explained there is a retry at the outer level; let me check where my comment went before rewriting.